src/unix/fs.c

```diff
@@ -136,12 +136,8 @@ static ssize_t uv__fs_fdatasync(uv_fs_t* req) {
 #if defined(__linux__) || defined(__sun) || defined(__NetBSD__)
   return fdatasync(req->file);
 #elif defined(__APPLE__)
-  /* Apple's fdatasync and fsync explicitly do NOT flush the drive write cache
-   * to the drive platters. This is in contrast to Linux's fdatasync and fsync
-   * which do, according to recent man pages. F_FULLFSYNC is Apple's equivalent
-   * for flushing buffered data to permanent storage.
-   */
-  return fcntl(req->file, F_FULLFSYNC);
+  /* See the comment in uv__fs_fsync. */
+  return uv__fs_fsync(req);
```

Comment:
Doesn't this emit a warning?  `uv__fs_fsync()` is declared below this function.

Reply:
Fixed. I didn't want to alias uv__fs_fsync to uv__fs_fdatasync on Apple in case someone might get the wrong idea we were doing something less than fsync.